src/librustdoc/formats/item_type.rs

```diff
@@ -48,7 +48,9 @@ crate enum ItemType {
     ProcAttribute = 23,
     ProcDerive = 24,
     TraitAlias = 25,
-    Generic = 26,
+    TyAssocType = 26,
+    TyAssocConst = 27,
```

Comment:
What's the difference exactly with `AssocType` and `AssocConst`?

Reply:
It's comparable to the difference between `TyMethod` and `Method`. The variants with prefix `Ty` always represent required trait items whereas the unprefixed versions correspond to provided trait items or impl items. The distinction is necessary in `render/mod.rs` and `print_item.rs` to generate correct hyperlinks.

On the JS side, the distinction does not matter though. E.g. it won't affect the search at all as they are treated the same in `search.js`: https://github.com/rust-lang/rust/blob/1081a0e36886d762823f2723bedfb9685df684ea/src/librustdoc/html/static/js/search.js#L597-L608

Reply:
I think it'd be better to remove this distinction from the JS side then. If we decide to not differentiate the anchors, it'll make the changes smaller too.

Reply:
I am afraid this is not possible. As I said, the `ItemType` enum on the Rust side has to be extended to properly implement this feature and as a result, the `itemTypes` array in `search.js` has to be updated, too, since they _have_ to be in a 1:1 correspondence per comments above `ItemType` and `itemTypes`. This is not harmful in the slightest because they are treated as being the same as I already mentioned.
The “tag” `"tymethod"` also has to exist on the JS side because of this forced bijection even though it's only ever used in the snippet I linked above. It's its only occurrence in JS.